xbmc/ServiceManager.cpp

```diff
@@ -72,24 +72,33 @@ bool CServiceManager::InitForTesting()
 
   m_databaseManager.reset(new CDatabaseManager);
 
-  m_fileExtensionProvider.reset(new CFileExtensionProvider());
-
+  m_binaryAddonManager.reset(new ADDON::CBinaryAddonManager());
   m_addonMgr.reset(new ADDON::CAddonMgr());
   if (!m_addonMgr->Init())
   {
     CLog::Log(LOGFATAL, "CServiceManager::%s: Unable to start CAddonMgr", __FUNCTION__);
     return false;
   }
 
+  if (!m_binaryAddonManager->Init())
```

Comment:
this needs databases afaict ? through the call to GetInstalledBinaryAddons -> CAddonMgr::GetInstalledAddons -> CAddonMgr::GetAddonsInternal -> AddonManager.cpp:587

Reply:
It needs add-on DB, which is created/updated when `m_databaseManager.reset(new CDatabaseManager);` is called above

Reply:
alright.

Reply:
This occurs in the constructor of DB manager. I considered breaking it out into an explicit InitializeAddonDB() function, but figured the extra code wasn't worth saving you a github comment ;)

Reply:
> to me it looked like you moved this to level 1 and db's were still at level 2.

`InitForTesting()` eschews levels for some reason

Reply:
there's good reasons for that but that was indeed what confused me.